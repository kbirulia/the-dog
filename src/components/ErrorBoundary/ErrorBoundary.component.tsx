import React, { Component, ReactNode } from "react";

interface Props {
  children: ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends Component<Props, State> {
  constructor(props: Props) {
    super(props);
    this.state = { hasError: false };
  }

  public static getDerivedStateFromError(): State {
    return { hasError: true };
  }

  public render(): ReactNode {
    const {
      state: { hasError },
      props: { children },
    } = this;
    if (hasError) {
      return <h1>Sorry.. there was an error</h1>;
    }

    return children;
  }
}

export default ErrorBoundary;
