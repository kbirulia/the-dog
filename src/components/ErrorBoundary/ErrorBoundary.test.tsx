import React from "react";
import renderer from "react-test-renderer";
import ErrorBoundary from "./ErrorBoundary.component";

const ErrorChild = () => {
  throw new Error("Component with error");
};

const Child = () => <div>Child</div>;

describe("ErrorBoundary", () => {
  test("should render an error", () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <ErrorChild />
        </ErrorBoundary>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("should render child", () => {
    const tree = renderer
      .create(
        <ErrorBoundary>
          <Child />
        </ErrorBoundary>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
