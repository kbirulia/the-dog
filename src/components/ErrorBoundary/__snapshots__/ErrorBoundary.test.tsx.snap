// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorBoundary should render an error 1`] = `
<h1>
  Sorry.. there was an error
</h1>
`;

exports[`ErrorBoundary should render child 1`] = `
<div>
  Child
</div>
`;
